src/protocol/identify/identify_msg_processor.cpp

```diff
@@ -42,13 +42,15 @@ namespace libp2p::protocol {
     // set an address of the other side, so that it knows, which address we used
     // to connect to it
     if (auto remote_addr = stream->remoteMultiaddr()) {
-      msg.set_observedaddr(std::string{remote_addr.value().getStringAddress()});
+      auto const &addr = remote_addr.value().getBytesAddress();
+      msg.set_observedaddr(std::string((const char*)addr.data(), addr.size()));
```

Comment:
I think it'd be the best to define a helper function for this operation, since it's done in a number of places and is not pretty to read.